frontend/views/Home/Admins.tsx

```diff
@@ -21,61 +51,165 @@ type AdminProps = {
  * @param {AdminProps} props
  * @returns {React.ReactElement} React component
  */
-export default function Admins(props: AdminProps): React.ReactElement {
-  const { token, admins, onRefresh } = props;
+function Admins(props: AdminProps): React.ReactElement {
+  const { user, theme, token, admins, onRefresh } = props;
+  const [deleteModal, setDeleteModal] = useState(false);
+  const [editModal, setEditModal] = useState(false);
+  const [selectedItem, setSelectedItem] = useState(null);
 
-  const onDelete = async (email: string) => {
+  const name = useRef();
+
+  const {
+    values,
+    handleChange,
+    handleBlur,
+    errors,
+    touched,
+    handleSubmit,
+    validateField,
+    setFieldValue,
+  } = useFormik({
+    validationSchema: EmailNameValidation,
+    initialValues: initialValues,
+    onSubmit: (values) => onEdit(values),
+  });
+
+  const onDelete = async () => {
     try {
-      await Admin.delete(email, token);
+      await Admin.delete(selectedItem.email, token);
     } catch {
       // show error message
     }
     onRefresh();
   };
 
-  const onEdit = (user: {
-    email: string;
-    name: string;
-    superUser: boolean;
-  }) => {
-    //TODO: update Admin.update() params
+  const onEdit = async (fields: EditFields) => {
+    const { name, email } = fields;
     try {
-      //Admin.update(email, token)
+      Admin.update(email, name, props.token);
+      setEditModal(false);
       onRefresh();
     } catch {
-      // show error message
+      // TODO: show error message
     }
   };
 
+  const superUserDeleteCheck = (item: any) => {
+    if (!item.superUser && item.email != user.email)
+      return (
+        <IconButton
+          icon="delete"
+          onPress={() => {
+            setDeleteModal(true);
+            setSelectedItem(item);
+          }}
+        />
+      );
+  };
+
+  const handleEditClick = (admin: Admin) => {
+    setFieldValue("name", admin.name);
+    setFieldValue("email", admin.email);
+    setEditModal(true);
+  };
+
   return (
-    <View style={{ flex: 1 }}>
-      <FlatList
-        data={admins}
-        keyExtractor={(_, index) => index.toString()}
-        renderItem={({ item }) => {
-          return (
+    <FlatList
+      style={{ flex: 1 }}
```

Comment:
In lined styles, also I think you should favor the previous format of wrapping the FlatList in a View. I think if you don't this may not display content on mobile